### PR TITLE
Materialize the Trends view

### PR DESCRIPTION
**Pull request by Jkrzy**
[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/603)

## What does this change?
Dropping the existing `trends` view and recreating as a [materialized view](https://www.postgresql.org/docs/10/rules-materializedviews.html). This persist the results or the trends analysis in a table-like form so it will no longer be calculated on page load. 

Also included is a mechanism by which the view can be refreshed via the `Trends` class and a management command.

With this PR, we'll be able to refresh the trends view via a management command, initially manually as a [cloudfoundry task](https://docs.cloudfoundry.org/devguide/using-tasks.html#run-tasks) w/ the CF CLI.

Something like:
```bash
cf run-task crt-portal-django "python manage.py refresh_trends" --name REFRESH_TRENDS
```

We should also look to run this periodically via CI by enhancing our circleCI `config.yml`.

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/migrations/0086_materialize_trends_view.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        row_number() OVER (PARTITION BY true) as id
+    FROM ts_stat($$
+        SELECT to_tsvector('english_simple', violation_summary)
+        FROM cts_forms_report
+        where
+            (create_date >= date_trunc('week', CURRENT_TIMESTAMP - interval '1 week'))
+    $$)
+    ORDER BY document_count DESC
+    LIMIT  10)
+    UNION
```
[the remark this answers is not in the document]
Comment (harikrishnagit) [reply crt_portal/cts_forms/migrations/0086_materialize_trends_view.py:28]: @Jkrzy How do I get to see the trends data/where is this data visible from?

[the remark this answers is not in the document]
Comment (Jkrzy) [reply crt_portal/cts_forms/migrations/0086_materialize_trends_view.py:28]: Good thought, there's absolutely likely clean-up that can be done here!

I'd love to first return functionality here with this PR, then re-evaluate the analysis, implementation, and display as part of the broader conversation on reporting. To that end, I intentionally avoided inspection/optimization of the SQL here, leaving it unmodified from its current state.

File: crt_portal/cts_forms/management/commands/refresh_trends.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class Command(BaseCommand):
+    help = 'Refreshes the Trends materialized view'
+
+    def handle(self, *args, **options):
+        start = time.time()
+        Trends.refresh_view()
+        end = time.time()
+        elapsed = round(end - start, 4)
+
+        self.stdout.write(self.style.SUCCESS(f'SUCCESS: Refreshed Trends view in {elapsed} seconds'))
```
[the remark this answers is not in the document]
Comment (Jkrzy) [reply crt_portal/cts_forms/management/commands/refresh_trends.py:15]: Any of the above! We're on the same page re: automation and scheduling, I made a few notes around that in the original issue. 

I have a PR in the works for adding to our circleCI pipeline w/ scheduling but wanted to push up this small piece first in the hopes of getting it up quickly and bringing the trends view back online.

File: crt_portal/cts_forms/migrations/0086_materialize_trends_view.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jkrzy) [reply crt_portal/cts_forms/migrations/0086_materialize_trends_view.py:28]: The trends view itself is made available at `/trends` within the cts_forms app, [configured here in urls.py](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/cts_forms/urls.py#L15),

[the remark this answers is not in the document]
Comment (harikrishnagit) [reply crt_portal/cts_forms/migrations/0086_materialize_trends_view.py:28]: > Good thought, there's absolutely likely clean-up that can be done here!
> 
> I'd love to first return functionality here with this PR, then re-evaluate the analysis, implementation, and display as part of the broader conversation on reporting. To that end, I intentionally avoided inspection/optimization of the SQL here, leaving it unmodified from its current state.

Makes sense, didn't know the history.

[the remark this answers is not in the document]
Comment (harikrishnagit) [reply crt_portal/cts_forms/migrations/0086_materialize_trends_view.py:28]: trends page (http://localhost:8000/form/trends/) does load up locally quickly. The improvement is probably distinguishable on stage or dev.

File: crt_portal/cts_forms/management/commands/refresh_trends.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cantsin) [reply crt_portal/cts_forms/management/commands/refresh_trends.py:15]: When we do scheduling, it'd be great if we could clear stale sessions from the database periodically. So we have more than one need for scheduling here. Maybe something to raise with the team at large.

**Pull request merged**